src/pages/iou/ReceiptSelector/NavigationAwareCamera.js

```diff
@@ -21,10 +28,56 @@ const defaultProps = {
     torchOn: false,
 };
 
+function useTabNavigatorFocus({cameraTabIndex, isInTabNavigator}) {
+    // Get navigation to get initial isFocused value (only needed once during init!)
+    const isPageFocused = useIsFocused();
+    const [isTabFocused, setIsTabFocused] = useState(false);
+
+    // Retrieve the animation value from the tab navigator, which ranges from 0 to the total number of pages displayed.
+    // Even a minimal scroll towards the camera page (e.g., a value of 0.001 at start) should activate the camera for immediate responsiveness.
+
+    // STOP!!!!!!! This is not a pattern to be followed! We are conditionally rendering this hook becase when used in the edit flow we'll never be inside a tab navigator.
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const tabPositionAnimation = isInTabNavigator ? useTabAnimation() : null;
+
+    useEffect(() => {
+        if (!tabPositionAnimation) {
+            return;
+        }
+        const index = Number(cameraTabIndex);
+
+        const listenerId = tabPositionAnimation.addListener(({value}) => {
+            // Activate camera as soon the index is animating towards the `cameraTabIndex`
+            requestAnimationFrame(() => {
+                setIsTabFocused(value > index - 1 && value < index + 1);
+            });
+        });
+
+        // We need to get the position animation value on component initialization to determine
```

Comment:
So in terms of react-navigation focus, all the tabs are focused as they are on the same screen. In order to know if the particular tab is focused we can use the tab position animation value.

Reply:
OK, I think I get it. When you say "tab position animation value", is that really just the "tab position value" (like an x/y coordinate)?

Reply:
It's literally `Animation.Value` between 0 and total_number_of_tabs-1

Reply:
Ah! OK, thanks for exaplaining that. Could you please improve this code comment to give more details like that? I suggest something like this:

> To determine if the tab is focused or not, `Animation.value` must be retrieved when the component mounts. `Animation.values` is a number between 0 and total_number_of_tabs - 1, and the only way to get the value in a synchronous way is to use a private method.

Which brings up another question... why does it have to be retrieved synchronously? Can't it be retrieved async?

Reply:
It's both. There is an event listener that listens for a value change and determines if the tab is in focus (so that's the async part), but there's also the sync part where the value is extracted on hook mount. This is because we need to know the value on component init to determine tab focus.